albatross/core/indexing.h

```diff
@@ -125,34 +99,39 @@ symmetric_subset(const std::vector<SizeType> &indices,
   return subset(indices, v.diagonal()).asDiagonal();
 }
 
-template <typename CovarianceType, typename SizeType>
-Distribution<CovarianceType> subset(const std::vector<SizeType> &indices,
-                                    const Distribution<CovarianceType> &dist) {
-  auto mean = subset(indices, Eigen::VectorXd(dist.mean));
-  if (dist.has_covariance()) {
-    auto cov = symmetric_subset(indices, dist.covariance);
-    return Distribution<CovarianceType>(mean, cov);
-  } else {
-    return Distribution<CovarianceType>(mean);
+/*
+ * Set a subset of an Eigen::Vector.  If this worked it'd be
+ * the equivalent of:
+ *
+ *     to[indices] = from;
+ */
+template <typename SizeType>
+inline void set_subset(const std::vector<SizeType> &indices,
+                       const Eigen::VectorXd &from, Eigen::VectorXd *to) {
```

Comment:
I *think* this aligns with the conventions we use elsewhere, namely if something is going to get modified in place it's a pointer, otherwise it's a const ref.

Reply:
Ah, I missed the bit in our C++ coding standard that said output parameters should be pointers. It makes sense, it usually forces the called to knowingly pass in a pointer.

It also strongly suggests using the return value for output instead of using a parameter when possible. `Eigen::VectorXd` not having efficient move semantics would be a reasonable motivation to not simply return them.